### ISSUE TITLE
DeserializationError when listing objects when s3 directory is empty

### ISSUE THREAD

**Issue by ghost** [diagnosia/flutter_aws_s3_client#6]
```
DeserializationError (Deserializing '[ListBucketResult, {Name: xxx, Prefix: mydir/, KeyCo...' to 'ListBucketResultParker' failed due to: Deserializing '[Name, xxx, Prefix, mydir/, KeyCount, 1, MaxKeys, 10...' to 'ListBucketResult' failed due to: Deserializing '[Key, mydir/, LastModified, 2020-06-12T14:18:55.000Z, ETag, "d41d8cd98f00b...' to 'BuiltList<Contents>' failed due to: Deserializing 'Key' to 'Contents' failed due to: type 'String' is not a subtype of type 'Iterable<dynamic>' in type cast)
```

The exception above is being thrown when I call 

```
await s3client.listObjects(prefix: 'mydir/', delimiter: '/');
```

when mydir is empty.

Works absolutely fine when the directory contains objects but not when it's empty.

Has anyone else seen this? Many thanks.

**Comment by knaeckeKami** [comment]
Thank you for this report. This should be fixed in version 0.5.1. If not, please let me know.

**Comment by ghost** [comment]
Thank you for your quick response and fix. Will try soon but trust the issue is gone :)
